src/lib/preferences.ts

```diff
@@ -182,6 +235,158 @@ function usePreferencesStore() {
+        keymapString +=
+          ' ' + createCodemirrorTypeKeymap(keymapItem.shortcutSecondStroke)
+      }
+      return keymapString
+    },
+    [keymap]
+  )
+
+  const updateKeymap = useCallback(
+    (
+      key: string,
+      firstShortcut: KeymapItemEditableProps,
+      secondShortcut?: KeymapItemEditableProps
+    ): Promise<void> => {
+      if (keymap == null) {
+        return Promise.reject('No keymap available')
+      }
+      if (findExistingShortcut(key, firstShortcut, keymap)) {
+        return Promise.reject('Existing keymap with the same shortcut')
+      }
+      const keymapItem = keymap.get(key)
+      if (keymapItem == null) {
+        return Promise.reject(`No such keymap to set for key: ${key}`)
+      }
+      keymap.set(key, {
+        description: keymapItem.description,
+        isMenuType: keymapItem.isMenuType,
+        shortcutMainStroke: {
+          ...keymapItem.shortcutMainStroke,
+          ...firstShortcut,
+        },
+        shortcutSecondStroke:
+          secondShortcut != null
+            ? {
+                ...keymapItem.shortcutSecondStroke,
+                ...secondShortcut,
+              }
+            : undefined,
+      })
+
+      setPreferences((preferences) => {
+        return {
+          ...preferences,
+          'general.keymap': keymap,
+        }
+      })
+
+      if (isMenuKeymap(keymapItem)) {
+        const { ipcRenderer } = window.require('electron')
+        ipcRenderer.send('menuAcceleratorChanged', [
+          key,
+          getMenuAcceleratorForKeymapItem(keymapItem),
+        ])
+      }
+      return Promise.resolve()
+    },
+    [keymap, setPreferences]
+  )
+
+  const removeKeymap = useCallback(
+    (key) => {
+      if (keymap == null) {
+        return false
+      }
+      const keymapItem = keymap.get(key)
+      if (keymapItem == null) {
+        return false
+      }
+      keymap.set(key, {
+        ...keymapItem,
+        shortcutMainStroke: undefined,
+        shortcutSecondStroke: undefined,
+      })
+      setPreferences((preferences) => {
+        return {
+          ...preferences,
+          'general.keymap': keymap,
+        }
+      })
+
+      if (isMenuKeymap(keymapItem)) {
+        const { ipcRenderer } = window.require('electron')
+        ipcRenderer.send('menuAcceleratorChanged', [key, null])
+      }
+      return true
+    },
+    [keymap, setPreferences]
+  )
+
+  const loadKeymaps = useCallback(() => {
+    const keymap = mergedPreferences['general.keymap']
+    if (keymap != null) {
+      const { ipcRenderer } = window.require('electron')
+      for (const [key, keymapItem] of keymap) {
+        if (isMenuKeymap(keymapItem)) {
+          ipcRenderer.send('menuAcceleratorChanged', [
+            key,
+            getMenuAcceleratorForKeymapItem(keymapItem),
+          ])
+        }
+      }
+    }
+  }, [mergedPreferences])
+
+  const resetKeymap = useCallback(() => {
+    if (keymap != null) {
+      keymap.clear()
```

Comment:
Default keymap should not be referenced else, since this would clear default keymap, and then this would not work.
So use the default keymap only here.